app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer
+
+import scala.collection.immutable.ArraySeq
+import scala.concurrent._
+
+import com.typesafe.scalalogging.StrictLogging
+import io.vertx.core.Vertx
+import io.vertx.core.http.{HttpMethod, HttpServer}
+import io.vertx.ext.web._
+import io.vertx.ext.web.handler.CorsHandler
+import sttp.tapir.server.vertx.VertxFutureServerInterpreter._
+
+import org.alephium.util.Service
+
+/** Stores AkkaHttp related instances created on boot-up */
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class ExplorerHttpServer(host: String, port: Int, val routes: ArraySeq[Router => Route])(
+    implicit val executionContext: ExecutionContext
+) extends Service
+    with StrictLogging {
+
+  private val vertx  = Vertx.vertx()
+  private val router = Router.router(vertx)
+  vertx
+    .fileSystem()
+    .existsBlocking(
+      "META-INF/resources/webjars/swagger-ui/"
+    ) // Fix swagger ui being not found on the first call
+  private val server = vertx.createHttpServer().requestHandler(router)
+
+  //scalastyle:off magic.number
+  router
+    .route()
+    .handler(
+      CorsHandler
+        .create(".*.")
+        .allowedMethod(HttpMethod.GET)
+        .allowedMethod(HttpMethod.POST)
+        .allowedMethod(HttpMethod.PUT)
+        .allowedMethod(HttpMethod.HEAD)
+        .allowedMethod(HttpMethod.OPTIONS)
+        .allowedHeader("*")
+        .allowCredentials(true)
+        .maxAgeSeconds(1800)
+    )
+
+  routes.foreach(route => route(router))
+
+  // scalastyle:on magic.number
+  private val httpBindingPromise: Promise[HttpServer] = Promise()
+
+  override def startSelfOnce(): Future[Unit] = {
+    for {
+      httpBinding <- server.listen(port, host).asScala
+    } yield {
+      logger.info(s"Listening http request on ${httpBinding.actualPort}")
+      httpBindingPromise.success(httpBinding)
+    }
+  }
+
+  /** Stop AkkaHttp server and terminates ActorSystem */
+  def stopSelfOnce(): Future[Unit] = {
+    for {
+      binding <- httpBindingPromise.future
+      _       <- binding.close().asScala
+    } yield {
+      logger.info(s"http unbound")
+    }
+  }
```

Comment:
Think there is a need close the Vertx instance here with `vertx.close()`? Without it the Thread is held

> [WARN] [10/03/2022 15:35:46.749] [default-akka.actor.internal-dispatcher-2] [CoordinatedShutdown(akka://default)] Coordinated shutdown phase [before-actor-system-terminate] timed out after 5000 milliseconds
2022-10-03 15:35:50,078 [-dispatcher-4] INFO  c.z.h.HikariDataSource                 - db.db - Shutdown completed.
2022-10-03 15:35:50,084 [-dispatcher-5] INFO  o.a.e.ExplorerHttpServer               - http unbound

With `vertx.close()` it properly exists

> 2022-10-03 15:37:41,909 [-dispatcher-6] INFO  c.z.h.HikariDataSource                 - db.db - Shutdown completed.
2022-10-03 15:37:41,936 [dispatcher-11] INFO  o.a.e.ExplorerHttpServer               - http unbound
Process finished with exit code 0

Reply:
oh nice catch, I didn't thought vertx  needed to be close!

Reply:
done